processor/src/main/java/org/mapstruct/ap/internal/model/common/Type.java

```diff
@@ -836,4 +848,35 @@ public boolean hasEmptyAccessibleContructor() {
 
         return null;
     }
+
+    public Type getElementType() {
+        if ( isArrayType() ) {
+            return getComponentType();
+        }
+        else {
+            return getTypeBound();
+        }
+    }
+
+    /**
+     * Searches for the given superclass and collects all type arguments for the given className.
+     * This is used for support with Java 8 streams, as must use the FQN in order to be able to compile
+     * with Java 6.
+     *
+     * @param superClassName the superclass or interface the generic type arguments are searched for
+     * @return a list of type arguments or null, if superclass was not found
+     */
+
+    public Collection<Type> determineTypeArguments(String superClassName) {
+        Class<?> superClass = null;
+        try {
+            superClass = Class.forName( superClassName );
```

Comment:
@gunnarmorling I had a look at the method that we were using before, and the `Class` was only needed to get it's name. Therefore, I did some changes in my method and moved checks there.